CMakeLists.txt

```diff
@@ -132,35 +130,43 @@ endif()
 
 option(INSTALL_PRIVATE_HEADERS "installs private headers in the same location as the public ones" OFF)
 
-find_package(BlocksRuntime QUIET)
-if(NOT BlocksRuntime_FOUND)
+if(NOT CMAKE_SYSTEM_NAME STREQUAL Darwin)
   set(BlocksRuntime_INCLUDE_DIR ${PROJECT_SOURCE_DIR}/src/BlocksRuntime)
 
   add_library(BlocksRuntime
-              STATIC
-                ${PROJECT_SOURCE_DIR}/src/BlocksRuntime/data.c
-                ${PROJECT_SOURCE_DIR}/src/BlocksRuntime/runtime.c)
+              ${PROJECT_SOURCE_DIR}/src/BlocksRuntime/data.c
+              ${PROJECT_SOURCE_DIR}/src/BlocksRuntime/runtime.c)
   set_target_properties(BlocksRuntime
                         PROPERTIES
                           POSITION_INDEPENDENT_CODE TRUE)
   if(HAVE_OBJC AND CMAKE_DL_LIBS)
-    set_target_properties(BlocksRuntime
-                          PROPERTIES
-                            INTERFACE_LINK_LIBRARIES ${CMAKE_DL_LIBS})
+    target_link_libraries(BlocksRuntime
+                          PUBLIC
+                            ${CMAKE_DL_LIBS})
   endif()
 
   add_library(BlocksRuntime::BlocksRuntime ALIAS BlocksRuntime)
 
+  if(ENABLE_SWIFT)
+    set(LIBDISPATCH_BLOCKSRUNTIME_HEADER_DIR ${SWIFT_LIBDIR}/swift/Block)
+  else()
+    set(LIBDISPATCH_BLOCKSRUNTIME_HEADER_DIR ${CMAKE_INSTALL_INCLUDEDIR})
+  endif()
+
   install(FILES
             ${PROJECT_SOURCE_DIR}/src/BlocksRuntime/Block.h
           DESTINATION
-            "${INSTALL_BLOCK_HEADERS_DIR}")
+            ${LIBDISPATCH_BLOCKSRUNTIME_HEADER_DIR})
```

Comment:
Header dir vs headers dir, let's keep this consistent. Also do we need to drop this cache var in this PR? Also I thought project-prefixed UPPER_SNAKE_CASED variables were conventionally used for cache vars rather than locals?

Reply:
Yeah, lowercase variables are preferred.